crates/drainer/src/errors.rs

```diff
@@ -8,16 +8,18 @@ pub enum DrainerError {
     #[error("Error during redis operation : {0}")]
     RedisError(error_stack::Report<redis::errors::RedisError>),
     #[error("Application configuration error: {0}")]
-    ConfigurationError(config::ConfigError),
+    ApplicationError(config::ConfigError),
```

Comment:
This is not exactly an application error, please name this variant appropriately.